### PR TITLE
Add the POST /store/coupon and GET /store/coupon/create handlers.

### PR DESCRIPTION
**Pull request by cdepillabout**
If you run the api, and then access http://localhost/store/coupon/create you will be able to create a new coupon.

This implements two of the handlers from #77.

@arowM Please review.

### CODE REVIEW

File: frontend/src/pug/storeUser_store_coupon_id_edit.pug [hunk 1]
```diff
@@ -105,16 +107,16 @@ include _layout
                 input#giftReferencePrice(
                   name="giftReferencePrice" type="number"
                   min="0"
-                  value="30"
+                  value!='\#{fromMaybe "30" giftReferencePrice}'
```
Comment (arowM) [inline frontend/src/pug/storeUser_store_coupon_id_edit.pug]: The "30" I injected is just a sample.
If you do not have explicit reason to use "30" as a default value, please use `mempty` as default.
(Please fix line 117, 144, 153 as well as this line)

**Comment by arowM** [comment]
Thanks for renaming to readable variable name and fixing specification documents.

When I tried to create a coupon with `validFrom` on Chrome, the page show me an error bellow.

```
could not parse: `2016-12-14'
```

Even if I do not specify `validFrom` and `validUntil`, the page show me an error bellow.

```
could not parse: `'
```

Also, I've created extra issue related to this PR on #83 and #85.

**Review by arowM — CHANGES_REQUESTED** [review]
@cdepillabout 
Thanks for renaming to readable variable name and fixing specification documents.

When I tried to create a coupon with `validFrom` on Chrome, the page show me an error bellow.

```
could not parse: `2016-12-14'
```

Even if I do not specify `validFrom` and `validUntil`, the page show me an error bellow.

```
could not parse: `'
```

Also, I've created extra issue related to this PR on #83 and #85.

**Comment by cdepillabout** [comment]
@arowM 
> When I tried to create a coupon with `validFrom` on Chrome, the page show me an error bellow.
>
> ```
> could not parse: `2016-12-14'
> ```

This has been fixed in commit ced7968.

> Even if I do not specify validFrom and validUntil, the page show me an error bellow.
>
> ```
> could not parse: `'
> ```

This has been fixed in commit 87b579c.

File: frontend/src/pug/storeUser_store_coupon_id_edit.pug (hunk 1 above)
Comment (cdepillabout) [reply frontend/src/pug/storeUser_store_coupon_id_edit.pug]: This is fixed in commit d289dfb.

**Comment by cdepillabout** [comment]
I'll go ahead and merge this in.

**Pull request merged**